client/websocket/websocket.go

```diff
@@ -303,19 +309,17 @@ func loadMarket(s *Server, cl *wsClient, req *marketLoad) (*bookFeed, *msgjson.E
 		return nil, msgjson.NewError(msgjson.RPCOrderBookError, errMsg)
 	}
 
+	cl.shutDownFeed()
+
 	cl.feedMtx.Lock()
```

Comment:
nACK.  This introduces an extra mutex lock/unlock cycle, and makes the feed shutdown and reassignment no longer atomic.
Avoiding three lines of code duplication is a poor trade off with the loss of inefficiency and atomicity.
Also, the `return cl.feed, nil` is now outside of the critical section.

Reply:
> This introduces an extra mutex lock/unlock cycle

Didn't think this would make any difference since it's not in hot path.

> makes the feed shutdown and reassignment no longer atomic

Didn't really think this could result into issues here, since all requests by client are sequentially ordered; however that's a good point to keep in mind

> Also, the return cl.feed, nil is now outside of the critical section.

I should probably do self-reviews for my own PRs first ;)

Reply:
Updated to address stuff mentioned above, hopefully without introducing new issues.